rply/parsergenerator.py

```diff
@@ -175,44 +177,28 @@ def build(self):
         g.compute_first()
         g.compute_follow()
 
-        # win32 temp directories are already per-user
-        if os.name == "nt":
-            cache_file = os.path.join(
-                tempfile.gettempdir(),
-                "rply-%s-%s-%s.json" % (
-                    self.VERSION, self.cache_id, self.compute_grammar_hash(g)
-                )
-            )
-        else:
-            cache_file = os.path.join(
-                tempfile.gettempdir(),
-                "rply-%s-%s-%s-%s.json" % (
-                    self.VERSION,
-                    os.getuid(),
-                    self.cache_id,
-                    self.compute_grammar_hash(g)
-                )
+        cache_dir = AppDirs("rply").user_cache_dir
+        cache_file = os.path.join(
+            cache_dir,
+            "rply-%s-%s-%s.json" % (
```

Comment:
Fixed!

On Tue, Sep 1, 2015 at 7:23 AM, Donald Stufft <notifications@github.com>
wrote:

> In rply/parsergenerator.py
> <https://github.com/alex/rply/pull/43#discussion_r38408431>:
>
> > -                    self.VERSION, self.cache_id, self.compute_grammar_hash(g)
> > -                )
> > -            )
> > -        else:
> > -            cache_file = os.path.join(
> > -                tempfile.gettempdir(),
> > -                "rply-%s-%s-%s-%s.json" % (
> > -                    self.VERSION,
> > -                    os.getuid(),
> > -                    self.cache_id,
> > -                    self.compute_grammar_hash(g)
> > -                )
> > +        cache_dir = AppDirs("rply").user_cache_dir
> > +        cache_file = os.path.join(
> > +            cache_dir,
> > +            "rply-%s-%s-%s.json" % (
>
> You probably don't need to start this filename with "rply" since it's
> going to be in a directory called "rply", but it doesn't hurt to have it
> either.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex/rply/pull/43/files#r38408431>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084